mpi/MPIFileIO.py

```diff
@@ -0,0 +1,106 @@
+from mpi4py import MPI
+from spt3g import core
+import random
+
+class MPIFileReader(object):
+    '''
+    Do parallel I/O and processing across an MPI communicator. The style of
+    parallelism here is that each process in the communicator gets a file, reads
+    the file, and processes the file. Supports shared files, which are read
+    from the lead process and then broadcasted to the others before they read
+    their own.
+
+    The list of files is specified as a list of lists. For example,
+    ['offline_calibration.g3', ['0000.g3', '0002.g3', '0003.g3']]
+    will cause all nodes to process 'offline_calibration.g3' and then the
+    remaining files (in the parallel block denoted by the nested list)
+    will be read and processed by the remaining nodes in a distributed way.
+    '''
+    def __init__(self, mpicomm, files):
+        self.comm = mpicomm
+        self.files = files
+        self.cur_reader = None
+        self.broadcast = False
+    def nextreader(self):
+        nextfilelist = self.files.pop(0)
+        self.broadcast = False
+        if isinstance(nextfilelist, str):
+            if self.comm.rank == 0:
+                self.broadcast = True
+                self.cur_reader = core.G3Reader(nextfilelist)
+            else:
+                self.cur_reader = lambda fr: self.comm.bcast(None,root=0)
```

Comment:
This is confusing.  I'd use a single `comm.bcast` call below, set `broadcast = True` on all nodes, and just have this `cur_reader` function return `None` on all but the root node.

Reply:
That is much cleaner, changed.